bridge/jira/import.go

```diff
@@ -0,0 +1,600 @@
+				return
+			}
+		}
+		if searchIter.HasError() {
+			out <- core.NewImportError(searchIter.Err, "")
+		}
+	}()
+
+	return out, nil
+}
+
+// Create a bug.Person from a JIRA user
+func (self *jiraImporter) ensurePerson(
+	repo *cache.RepoCache, user User) (*cache.IdentityCache, error) {
+
+	// Look first in the cache
+	i, err := repo.ResolveIdentityImmutableMetadata(
+		keyJiraUser, string(user.Key))
+	if err == nil {
+		return i, nil
+	}
+	if _, ok := err.(entity.ErrMultipleMatch); ok {
+		return nil, err
+	}
+
+	i, err = repo.NewIdentityRaw(
+		user.DisplayName,
+		user.EmailAddress,
+		user.Key,
+		"",
+		map[string]string{
+			keyJiraUser: string(user.Key),
+		},
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	self.out <- core.NewImportIdentity(i.Id())
+	return i, nil
+}
+
+// Create a bug.Bug based from a JIRA issue
+func (self *jiraImporter) ensureIssue(
+	repo *cache.RepoCache, issue Issue) (*cache.BugCache, error) {
+	author, err := self.ensurePerson(repo, issue.Fields.Creator)
+	if err != nil {
+		return nil, err
+	}
+
+	b, err := repo.ResolveBugCreateMetadata(keyJiraID, issue.ID)
+	if err != nil && err != bug.ErrBugNotExist {
+		return nil, err
+	}
+
+	if err == bug.ErrBugNotExist {
+		cleanText, err := text.Cleanup(string(issue.Fields.Description))
+		if err != nil {
+			return nil, err
+		}
+
+		title := fmt.Sprintf("[%s]: %s", issue.Key, issue.Fields.Summary)
+		b, _, err = repo.NewBugRaw(
+			author,
+			issue.Fields.Created.Unix(),
+			title,
+			cleanText,
+			nil,
+			map[string]string{
+				keyOrigin:      target,
+				keyJiraID:      issue.ID,
+				keyJiraKey:     issue.Key,
+				keyJiraProject: self.conf[keyProject],
+			})
+		if err != nil {
+			return nil, err
+		}
+
+		self.out <- core.NewImportBug(b.Id())
+	}
+
+	return b, nil
+}
+
+// Return a unique string derived from a unique jira id and a timestamp
+func getTimeDerivedID(jiraID string, timestamp MyTime) string {
+	return fmt.Sprintf("%s-%d", jiraID, timestamp.Unix())
+}
+
+// Create a bug.Comment from a JIRA comment
+func (self *jiraImporter) ensureComment(
+	repo *cache.RepoCache, b *cache.BugCache, item Comment) error {
+	// ensure person
+	author, err := self.ensurePerson(repo, item.Author)
+	if err != nil {
+		return err
+	}
+
+	targetOpID, err := b.ResolveOperationWithMetadata(
+		keyJiraID, item.ID)
+	if err != nil && err != cache.ErrNoMatchingOp {
+		return err
+	}
+
+	// If the comment is a new comment then create it
+	if targetOpID == "" && err == cache.ErrNoMatchingOp {
+		var cleanText string
+		if item.Updated != item.Created {
+			// We don't know the original text... we only have the updated text.
+			cleanText = ""
+		} else {
+			cleanText, err = text.Cleanup(string(item.Body))
+			if err != nil {
+				return err
+			}
+		}
+
+		// add comment operation
+		op, err := b.AddCommentRaw(
+			author,
+			item.Created.Unix(),
+			cleanText,
+			nil,
+			map[string]string{
+				keyJiraID:      item.ID,
+				keyJiraProject: self.conf[keyProject],
```

Comment:
hmm, is it useful to tag operations with the project if the bug itself is already tagged ?

Reply:
I can't think of a reason... seems like it's not useful. I will remove the project tag here and in other operations.